libraries/ssh_crypto.rb

```diff
@@ -16,261 +16,123 @@
 # author: Dominik Richter
 # author: Patrick Muench
 
-class SshCrypto < Inspec.resource(1) # rubocop:disable Metrics/ClassLength
+class SshCrypto < Inspec.resource(1)
   name 'ssh_crypto'
 
-  def ssh_version
-    inspec.command('ssh -V 2>&1 | cut -f1 -d" " | cut -f2 -d"_"').stdout.to_f
+  CRYPTO ||= {
+    macs: {
+      6.6 => 'hmac-sha2-512-etm@openssh.com,hmac-sha2-256-etm@openssh.com,umac-128-etm@openssh.com,hmac-sha2-512,hmac-sha2-256',
+      5.9 => 'hmac-sha2-512,hmac-sha2-256,hmac-ripemd160',
+      5.3 => 'hmac-ripemd160,hmac-sha1',
+    },
+    ciphers: {
+      5.3 => 'aes256-ctr,aes192-ctr,aes128-ctr',
+      6.6 => 'chacha20-poly1305@openssh.com,aes256-gcm@openssh.com,aes128-gcm@openssh.com,aes256-ctr,aes192-ctr,aes128-ctr',
+    },
+    kex: {
+      6.6 => 'curve25519-sha256@libssh.org,diffie-hellman-group-exchange-sha256',
+      5.9 => 'diffie-hellman-group-exchange-sha256',
+    }
+  }.freeze
+
+  PRIVILEGE_SEPARATION ||= {
+    5.3 => 'yes',
+    5.9 => 'sandbox',
+  }.freeze
+
+  HOSTKEY_ALGORITHMS ||= {
+    5.3 => %w[rsa],
+    6.0 => %w[rsa ecdsa],
+    6.6 => %w[rsa ecdsa ed25519],
+  }.freeze
+
+  FALLBACK_SSH_VERSION ||= 5.9
+
+  def real_ssh_version 
```

Comment:
I guess we need that only as a intermediate step to avoid breaking the baseline and do not need it when everything is complete, right? (I see this command as part of ssh_version)

Reply:
I hope so yes. This is indeed only temporary to avoid any breaking changes to the baseline.